WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -76,9 +76,49 @@ private void StepInputRegistrationPhase()
 				}
 				else
 				{
-					round.SetPhase(Phase.ConnectionConfirmation);
+					var areAllTxoUnspent = await CheckTxoSpendStatusAsync(round, cancel).ConfigureAwait(false);
+					if (areAllTxoUnspent)
+					{
+						round.SetPhase(Phase.ConnectionConfirmation);
+					}
+					else
+					{
+						round.SetPhase(Phase.Ended);
+						round.LogInfo($"The round was disrupted by users who spent their coins.");
```

Comment:
Here, double spending can be handled by just removing the offending inputs, since they aren't confirmed yet they don't have any credentials yet so the round is still recoverable.

The same check should probably repeated for the other phase transitions, in which case it really should fail because it's too late if credentials were issued to the offending input's owner. When double spending is detected later, I suppose a blame round would also make sense, perhaps without requiring signatures? i.e. whitelist all inputs apart from double spent ones.

Secondly, I think although Wasabi doesn't care about the reason why a round failed, maybe at the protocol level we should take this into consideration?

Reply:
I remove the offending alices now but I don't perform the same check for other phase transitions. I would like to do it in the next PR. 

> Secondly, I think although Wasabi doesn't care about the reason why a round failed, maybe at the protocol level we should take this into consideration?

I don't know what you mean, could you explain please?

Reply:
Re the first thing, I mean specifically if it's caught at this phase, after removing them the round can still proceed to connection confirmation phase, it's not too late so it doesn't have to go directly to `Phase.Ended`.

About the second thing, with auto coinjoin behavior this kind of failure is not user facing so it doesn't matter too much, but if we think about it more generally (manual coinjoins, other clients potentially) I think it makes sense to convey the reason the round is being aborted to the client, but I'm not sure about the details, I think just something to think about.